src/main/java/com/qubit/solution/fenixedu/integration/cgd/domain/logs/CgdCommunicationLog.java

```diff
@@ -54,6 +54,12 @@ public static Optional<CgdCommunicationLog> findLatestStudentLog(Person person)
         return findLogsByStudent(person).stream().findFirst();
     }
 
+    @Atomic
+    @Override
+    public void setSearchDate(DateTime searchDate) {
```

Comment:
@miguel-morais let's create a simpler API to anyone that uses CgdCommunicationLog.  Instead of having them to know that they have to perform setters directly let's create domain methods that now what to do:

I would have updateSearchDate() and updateSendDate() that would then set the setSerarchDate and setSendDate to DateTime.now() and those would be the @Atomic methods.